packages/components/modules/messages/CreateGroup/index.tsx

```diff
@@ -0,0 +1,356 @@
+import { useChatRoom } from '../context'
+import { useCreateChatRoomMutation } from '../graphql/mutations/CreateChatRoom'
+import DefaultProfileCard from './ProfileCard'
+import {
+  DEFAULT_FORM_VALIDATION,
+  DEFAULT_FORM_VALUES,
+  DEFAULT_IMAGE_FORMATS,
+  DEFAULT_IMAGE_MAX_SIZE,
+  FORM_VALUE,
+} from './constants'
+import { HeaderContainer, SearchbarContainer, UploadProfileContainer } from './styled'
+import { CreateGroupProps } from './types'
+import { getImageUrl } from './utils'
+
+const CreateGroup: FC<CreateGroupProps> = ({
+  allProfilesRef,
+  // TODO: Check if this is the correct way to get the remotePatternsHostName
+  remotePatternsHostName = process.env.NEXT_PUBLIC_REMOTE_PATTERNS_HOSTNAME || '', // To get this working in staging/prod, change NEXT_PUBLIC_REMOTE_PATTERNS_HOSTNAME to the media host being used (e.g.: digitalocean, aws, etc)
+  ProfileCard = DefaultProfileCard,
+  ProfileCardProps = {},
+  Searchbar = DefaultSearchbar,
+  SearchbarProps = {},
+  VirtuosoProps = {},
+  setIsInGroupChatCreation,
+  setIsInExistingChatRoomsView,
+}) => {
+  const theme = useTheme()
+  const {
+    data: { allProfiles },
+    loadNext,
+    isLoadingNext,
+    hasNext,
+    refetch,
+  } = useAllProfilesList(allProfilesRef)
+  const [isPending, startTransition] = useTransition()
+  const {
+    control: searchControl,
+    reset: searchReset,
+    watch: searchWatch,
+  } = useForm({ defaultValues: { search: '' } })
+
+  const {
+    control,
+    setValue,
+    watch,
+    getFieldState,
+    clearErrors,
+    reset,
+    handleSubmit,
+    formState: { isValid, isDirty, dirtyFields },
+  } = useForm({
+    defaultValues: DEFAULT_FORM_VALUES,
+    resolver: zodResolver(DEFAULT_FORM_VALIDATION),
+    mode: 'onBlur',
+  })
+
+  const { currentProfile } = useCurrentProfile()
+
+  const [commit, isMutationInFlight] = useCreateChatRoomMutation()
+  const { setChatRoom } = useChatRoom()
+
+  const onSubmit = handleSubmit((data: any) => {
+    const dirtyValues = filterDirtyValues({ values: data, dirtyFields })
+    const { title, participants, image } = data
+    const participantsIds = participants.map((member: any) => member.id)
+    const uploadables: { image?: File | Blob } = {}
+    if ('image' in dirtyValues && image && typeof image !== 'string') {
+      uploadables.image = image
+    }
+
+    commit({
+      variables: {
+        input: {
+          title,
+          isGroup: true,
+          profileId: currentProfile?.id as string,
+          participants: participantsIds,
+        },
+        connections: [
+          // TODO: add filter handling (for now we can default 'unreadMessages' to false)
+          ConnectionHandler.getConnectionID(currentProfile?.id as string, 'roomsList_chatRooms', {
+            unreadMessages: false,
+          }),
+        ],
+      },
+      uploadables,
+      onCompleted: (response) => {
+        setChatRoom({ id: response?.chatRoomCreate?.room?.node?.id })
```

Comment:
Do we need some error handling here (as for example in apps/web/app/(with-navigation)/user/settings/AccountSettings/AccountProfile/index.tsx)?

Reply:
Nice catch I'll update it, thanks